core/server/guilds.js

```diff
@@ -79,27 +102,35 @@ Meteor.methods({
     });
     analytics.updateGuild(Guilds.findOne(guildId), {}, Meteor.userId());
 
+    emitMemberEvent(guildId);
     log('addGuildUsers: done');
   },
-  removeTeamUser(guildId, userId) {
-    check([guildId, userId], [Match.Id]);
-    log('removeTeamUser: start', { guildId, userId });
+  removeTeamUsers(guildId, userIds) {
+    check(guildId, Match.Id);
+    check(userIds, [Match.Id]);
+    log('removeTeamUsers: start', { guildId, userIds });
 
+    if (!userIds.length) return;
     if (!this.userId) throw new Meteor.Error('not-authorized', 'User not allowed');
 
-    if (!canEditGuild(Meteor.user(), Guilds.findOne(guildId))) throw new Meteor.Error('not-authorized', `Missing permissions to edit team members`);
+    const guild = Guilds.findOne(guildId);
+    if (!canEditGuild(Meteor.user(), guild)) throw new Meteor.Error('not-authorized', `Missing permissions to edit team members`);
 
-    const user = Meteor.users.findOne(userId);
-    if (user.guildId !== guildId) throw new Meteor.Error('user-invalid', 'Given user is not in the team');
+    let users = Meteor.users.find({ _id: { $in: userIds }, guildId });
+    if (users.length === 0) throw new Meteor.Error('user-invalid', 'Given user is not in the team');
 
-    Meteor.users.update(userId, { $unset: { guildId: 1 } });
+    Meteor.users.update({ _id: { $in: userIds } }, { $unset: { guildId: 1 } });
 
     // analytics
-    analytics.identify(Meteor.users.findOne(userId));
-    analytics.track(this.userId, '👨‍👨‍👦 Guild Remove User', { user_id: userId, guild_id: guildId });
-    analytics.updateGuild(Guilds.findOne(guildId), {}, userId);
+    users = Meteor.users.find({ _id: { $in: userIds } }).fetch();
+    users.forEach(currentUser => {
+      analytics.identify(Meteor.users.findOne(currentUser));
```

Comment:
You can use the `currentUser` variable here and rename it to `removedUser` maybe